src/class/cdc/cdc_device.c

```diff
@@ -425,6 +425,9 @@ bool cdcd_xfer_cb(uint8_t rhport, uint8_t ep_addr, xfer_result_t result, uint32_
         usbd_edpt_xfer(TUD_OPT_RHPORT, p_cdc->ep_in, NULL, 0);
       }
     }
+
+    // invoke transmit callback to possibly refill tx fifo
+    if ( tud_cdc_tx_cb && !tu_fifo_full(&p_cdc->tx_ff) ) tud_cdc_tx_cb(itf);
```

Comment:
1. This should be called after the above flush, else we could send ZLP while this fill up the fifo which is not ideal.
2. To make it clear, we should rename the callback to `tud_cdc_tx_complete_cb()`